### PR TITLE
METRON-2074: Script to handle TGT renewal with Storm and Kerberos enabled

### PR DESCRIPTION
**Pull request by mmiklavc**
## Contributor Comments

https://issues.apache.org/jira/browse/METRON-2074

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- n/a Have you written or updated unit tests and or integration tests to verify your changes?
- n/a If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

### CODE REVIEW

**Comment by mmiklavc** [comment]
# Test Instructions

1. Run up full dev
1. Setup Kerberos per the instructions in https://github.com/apache/metron/blob/master/metron-deployment/Kerberos-ambari-setup.md
1. Modify the maxlife and maxrenewlife for your TGT. Here we're setting it to 3 and 4 minutes, respectively, but you can set the values even shorter for testing purposes.
    ```
    kadmin.local -q "modprinc -maxlife 3minutes krbtgt/EXAMPLE.COM@EXAMPLE.COM"
    kadmin.local -q "modprinc -maxrenewlife 4minutes krbtgt/EXAMPLE.COM@EXAMPLE.COM"
    ```
1. Restart the bro parser topology. We'll focus on one topology for this exercise.
1. Allow your max renew time to pass. If you look at your Storm bolt logs, you should see at least one renewal prior to expiration, e.g.
    ```
    2018-09-17 06:52:54.506 o.a.s.s.o.a.z.Login Thread-1 [INFO] Initiating logout for metron@EXAMPLE.COM
    2018-09-17 06:52:54.507 o.a.s.s.o.a.z.Login Thread-1 [INFO] Initiating re-login for metron@EXAMPLE.COM
    2018-09-17 06:52:54.521 o.a.s.s.o.a.z.Login Thread-1 [INFO] TGT valid starting at:        Mon Sep 17 06:53:17 UTC 2018
    2018-09-17 06:52:54.527 o.a.s.s.o.a.z.Login Thread-1 [INFO] TGT expires:                  Mon Sep 17 07:03:17 UTC 2018
    2018-09-17 06:52:54.527 o.a.s.s.o.a.z.Login Thread-1 [INFO] TGT refresh sleeping until: Mon Sep 17 07:01:29 UTC 2018
    ```
1. You should also start to see error messages like the following in the Nimbus logs:
    ```
    2018-09-17 07:12:52.064 o.a.s.s.a.k.AutoTGT timer [INFO] Renewing TGT for metron@EXAMPLE.COM
    2018-09-17 07:12:52.064 o.a.s.s.a.k.AutoTGT timer [WARN] Failed to refresh TGT javax.security.auth.RefreshFailedException: This ticket is past its last renewal time.
    ```
1. Run the tgt_renew.py script per the Kerberos README instructions in this PR and verify that you again start to see TGT login/renewal for metron@EXAMPLE.COM

**Comment by mmiklavc** [comment]
Appears to be an intermittent test failure. I've logged it in this Jira - https://issues.apache.org/jira/browse/METRON-2077. Re-running Travis.

File: metron-deployment/Kerberos-manual-setup.md [hunk 1]
```diff
@@ -636,3 +637,43 @@ The random access indexer topology fails with the following exception.  This exc
...
+Apache Storm doesn't handle automatic TGT ticket renewal for their running topologies. Instead, it is left up to the operations team deploying the Storm topologies
+in a Kerberized environment to manage this themselves. We've included a Python script that can be setup with a cron process to automatically manage the renewal
+process for you. The script should be run on an interval that is shorter than the renew_lifetime configured for your TGT.
+
+### Setup Instructions
+
+Run the following on a node with a Storm and Metron client installed. We need python 2.7 via virtualenv for this to work correctly.
+
+```
+su - metron
```
[the remark this answers is not in the document]
Comment (mmiklavc) [reply metron-deployment/Kerberos-manual-setup.md]: I've updated the docs based on your feedback. How's it look now?

[the remark this answers is not in the document]
Comment (anandsubbu) [reply metron-deployment/Kerberos-manual-setup.md]: +1 looks great!

**Pull request closed without merging**